### PR TITLE
feat(#major); optimism-gateway; Add Optimism Gateway subgraph

### PR DESCRIPTION
**Pull request by spitko**
- Pool id is set to L1 token address on both Optimism and Mainnet networks
- Protocol id is set to (L1/L2)StandardBridge contract address
- BridgeMessages includes internal BridgeTransfer messages
- BridgeMessages sent from Mainnet before `L1CrossDomainMessenger` contract upgrade are not included 
- Only sent messages are tracked, because there is not enough data included for received messages (just message hash)
- Some early transfers on Optimism are missing price data, because price oracles were not deployed yet

https://okgraph.xyz/?q=spitko%2Foptimism-gateway-ethereum
https://okgraph.xyz/?q=spitko%2Foptimism-gateway-optimism

### CODE REVIEW

**Comment by spitko** [comment]
close #1477

**Comment by bye43** [comment]
@dmelotik The ETH deployment for this passed QA here - https://github.com/messari/subgraphs/issues/1700. Can you deploy it on the messari-hosted service when you get the chance and mark it as `prod`?

**Comment by dmelotik** [comment]
@spitko can you resolve the deployment.json conflict? This is the PR that did it: https://github.com/messari/subgraphs/pull/1724

The optimism gateway metadata is already in `deployment.json`. But like @bye43 said can you mark it as `prod` now?

File: subgraphs/optimism-bridge-v2/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+- Received messages are not being tracked, because there is not enough data included in the event.
+
+- Some of the earliest transfers on Optimism network are missing price data, because price oracles were not deployed yet
+
+## Smart Contract Interactions
+
+![Optimism Gateway](../../docs/images/protocols/optimism-gateway.png "Optimism Gateway")
+
+## Links
```
Comment (dmelotik) [inline subgraphs/optimism-bridge-v2/README.md:57]: Could you also add a link to the source of the contract addresses? I am finding this:https://community.optimism.io/docs/useful-tools/networks/#optimism-mainnet

File: subgraphs/optimism-bridge-v2/src/mappings/L1StandardBridge.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    Address.fromString(ETH_ADDRESS_OPTIMISM),
+    CrosschainTokenType.WRAPPED,
+    ethAddress
+  );
+  pool.addDestinationToken(crossToken);
+
+  const acc = sdk.Accounts.loadAccount(event.params._from);
+  acc.transferOut(
+    pool,
+    pool.getDestinationTokenRoute(crossToken)!,
```
Comment (dmelotik) [inline subgraphs/optimism-bridge-v2/src/mappings/L1StandardBridge.ts:109]: should we be doing any error handling here? Is it possible this could be null?

**Review by dmelotik — COMMENTED** [review]
Great work @spitko !

Some questions/comments to help my understanding:
- The token list of tokens supported is ~300 assets, why does the subgraph have only 134? https://github.com/ethereum-optimism/ethereum-optimism.github.io/blob/master/optimism.tokenlist.json

**Comment by dmelotik** [comment]
LGTM! I am going to let @jaimehgb do a final approve since I haven't done too many bridge reviews.

**Review by dmelotik — APPROVED** [review]
approving LGTM

**Pull request merged**